bin/propolis-server/src/lib/migrate/source.rs

```diff
@@ -310,22 +391,72 @@ impl<T: AsyncRead + AsyncWrite + Unpin + Send> SourceProtocol<T> {
     async fn run(&mut self) -> Result<(), MigrateError> {
         info!(self.log(), "Entering Source Migration Task");
 
-        self.run_phase(MigratePhase::MigrateSync).await?;
-        self.run_phase(MigratePhase::RamPushPrePause).await?;
-        self.run_phase(MigratePhase::Pause).await?;
-        self.run_phase(MigratePhase::RamPushPostPause).await?;
-        self.run_phase(MigratePhase::TimeData).await?;
-        self.run_phase(MigratePhase::DeviceState).await?;
-        self.run_phase(MigratePhase::RamPull).await?;
-        self.run_phase(MigratePhase::ServerState).await?;
-        self.run_phase(MigratePhase::Finish).await?;
-
-        info!(self.log(), "Source Migration Successful");
-        Ok(())
+        let result: Result<(), MigrateError> = async {
```

Comment:
why is the type annotation necessary here? Don't all the calls to `run_phase` return the same type?

Reply:
They do, but for some reason the compiler needs a hint as to the error type. (It's fine with `_` as the success type.)

Reply:
weird. that's fine, i guess.